lib/wallet/http.js

```diff
@@ -2164,6 +1752,41 @@ class HTTP extends Server {
       return null;
     });
   }
+
+  /**
+   * Registers a new http handler at provided path
+   * the handler method is passed regular req, res along
+   * with a context object that holds vallet specific
+   * objects that might be used to extend wallet plugin
+   * functionality
+   *
+   * @param {*} httpMethod string value of one of
+   * 'post', 'get', 'put', 'patch', 'delete'
+   * @param {*} path location handler is listening
+   * @param {*} handler http handler that will handle requests
+   * arriving to `path` registered
+   * @param {Object} context context
+   * @returns {undefined}
+   */
+  registerRoute(httpMethod, path, handler, context) {
+    const validHttpMethods = ['post', 'get', 'put', 'patch', 'del'];
+    assert(httpMethod && typeof httpMethod === 'string',
+    'httpMethod parameter must be provided and it must be a string');
+    assert(validHttpMethods.includes(httpMethod.toLowerCase()),
+    'httpMethod must be one of ' + validHttpMethods.join(','));
+
+    // update context
+    context.TransactionOptions = TransactionOptions;
+    context.Validator = Validator;
+    context.network = this.network;
+    context.rules = rules;
```

Comment:
Some of these context items aren't needed since they can just be imported later, right?

Reply:
Except below 3 we can import the others, right.

context.TransactionOptions = TransactionOptions;
context.Validator = Validator;
context.network = this.network;